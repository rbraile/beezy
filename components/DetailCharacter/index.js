import React from 'react'
import Fields from '../Fields'

const DetailCharacter = ({ character }) => {
  const detailList = [
    'name',
    'gender',
    'height',
    'hair_color',
    'skin_color',
    'eye_color',
    'birth_year',
    'mass',
  ]
  return character.loading ? (
    <Spinner />
  ) : (
    <Fields detailList={detailList} details={character.detail} />
  )
}

export default DetailCharacter
