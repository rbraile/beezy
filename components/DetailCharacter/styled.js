import styled from 'styled-components'

export const Container = styled.div`
  text-align: center;
  margin-bottom: 20px;

  h2 {
    background: #f2f2f2;
  }
`
