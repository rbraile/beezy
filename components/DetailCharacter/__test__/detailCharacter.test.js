import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'

import DetailCharacter from '../index'

describe('TEST = CHARACTERLIST', () => {
  const props = {
    character: {
      loading: false,
      detail: {
        name: 'Anakin Skywalker',
        height: '188',
        mass: '84',
        hair_color: 'blond',
        skin_color: 'fair',
        eye_color: 'blue',
        birth_year: '41.9BBY',
        gender: 'male',
      },
    },
  }

  test('TEST = CHARACTERLIST = SNAPSHOT', () => {
    const wrapper = shallow(<DetailCharacter {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('TEST = CHARACTERLIST = MOUNT ', () => {
    const wrapper = mount(<DetailCharacter {...props} />)
    expect(wrapper.find('p')).toHaveLength(8)
  })
})
