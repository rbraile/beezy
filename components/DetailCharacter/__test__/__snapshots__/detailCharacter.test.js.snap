// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST = CHARACTERLIST TEST = CHARACTERLIST = SNAPSHOT 1`] = `
<Fields
  detailList={
    Array [
      "name",
      "gender",
      "height",
      "hair_color",
      "skin_color",
      "eye_color",
      "birth_year",
      "mass",
    ]
  }
  details={
    Object {
      "birth_year": "41.9BBY",
      "eye_color": "blue",
      "gender": "male",
      "hair_color": "blond",
      "height": "188",
      "mass": "84",
      "name": "Anakin Skywalker",
      "skin_color": "fair",
    }
  }
/>
`;
